docs/lifecycle.md

```diff
@@ -117,35 +154,87 @@ It is extended for lifecycle management, to maintain a list of buckets
 which have to be processed for lifecycle actions, so that only those
 buckets are processed.
 
-The list of buckets is stored in zookeeper nodes with the path:
+The list of buckets is stored in Zookeeper nodes with the path:
 
 ```
-[/chroot_path]/lifecycle/data/buckets/${ownerId}:${bucket}
+[/chroot_path]/lifecycle/data/buckets/${ownerId}:${bucketUID}:${bucketName}
 ```
 
 * **ownerId** is the canonical ID of the bucket owner
-* **bucket** is the bucket name
 
-The node value is not used (for now at least), so it's set to `null`
-(the JSON `null` value stored as a binary string).
+* **bucketUID** is the globally unique ID assigned to a bucket at
+    creation time (avoids conflicts with buckets re-created with the
+    same name)
+
+* **bucketName** is the bucket name
+
+The node value is not meaningful, so it's set to `null` (the JSON
+`null` value stored as a binary string).
 
 More details on actions taken when specific entries are processed from
 the metadata log:
 
 * when processing a log entry that is an update of bucket attributes,
   look at whether any lifecycle action is enabled for that bucket:
-  * if any action is enabled, create the zookeeper node for this
-    owner/bucket (ignore if already exists)
-  * if no action is enabled, delete the zookeeper node for this
-    owner/bucket (ignore if not found)
+
+  * if any action is enabled, create the Zookeeper node (ignore if
+    already exists)
+
+  * if no action is enabled, delete the Zookeeper node (ignore if not
+    found)
+
 * when processing a log entry that is a deletion of a bucket, delete
-  the zookeeper node for that owner/bucket (ignore if not found).
+  the Zookeeper node (ignore if not found).
+
+The resulting Zookeeper list should contain a current view of all
+buckets that have an attached and enabled lifecycle configuration, up
+to the point where the metadata log has been processed.
+
+Note that in the future, we may request the MongoDB database to get
+the list of buckets with lifecycle configuration instead of using
+Zookeeper nodes.
```

Comment:
What would prevent us from implementing this MongoDB usage off-the-bat?

Reply:
We need the proper clients to be configured to query mongodb directly and extract the full list of buckets from there.

I recall the zookeeper list method has been implemented to only consider buckets that have lifecycle attached to them for efficiency, but I am thinking this is not that useful and brings more complexity that could be removed eventually.